frontend/appflowy_flutter/lib/plugins/document/presentation/editor_plugins/mention/mention_date_block.dart

```diff
@@ -62,48 +75,71 @@ class MentionDateBlock extends StatelessWidget {
         builder: (context, appearance) =>
             BlocBuilder<ReminderBloc, ReminderState>(
           builder: (context, state) {
-            final reminder =
-                state.reminders.firstWhereOrNull((r) => r.id == reminderId);
-            final noReminder = reminder == null && isReminder;
+            final reminder = state.reminders
+                .firstWhereOrNull((r) => r.id == widget.reminderId);
+            final noReminder = reminder == null && widget.isReminder;
 
             final formattedDate = appearance.dateFormat
                 .formatDate(parsedDate!, includeTime, appearance.timeFormat);
 
+            final timeStr = parsedDate != null
+                ? _timeFromDate(parsedDate!, appearance.timeFormat)
+                : null;
+
             final options = DatePickerOptions(
-              selectedDay: parsedDate,
               focusedDay: parsedDate,
-              firstDay: isReminder
+              popoverMutex: mutex,
+              selectedDay: parsedDate,
+              firstDay: widget.isReminder
                   ? noReminder
                       ? parsedDate
                       : DateTime.now()
                   : null,
               lastDay: noReminder ? parsedDate : null,
+              timeStr: timeStr,
               includeTime: includeTime,
+              enableRanges: false,
+              dateFormat: appearance.dateFormat,
               timeFormat: appearance.timeFormat,
               onIncludeTimeChanged: (includeTime) {
+                this.includeTime = includeTime;
                 _updateBlock(parsedDate!.withoutTime, includeTime);
 
                 // We can remove time from the date/reminder
                 // block when toggled off.
-                if (isReminder) {
+                if (widget.isReminder) {
                   _updateScheduledAt(
-                    reminderId: reminderId!,
+                    reminderId: widget.reminderId!,
                     selectedDay:
                         includeTime ? parsedDate! : parsedDate!.withoutTime,
                     includeTime: includeTime,
                   );
                 }
               },
-              onDaySelected: (selectedDay, focusedDay, includeTime) {
-                parsedDate = selectedDay;
+              onStartTimeChanged: (time) {
```

Comment:
What if `time` is invalid, for example: `1200` or `13.00`? The user will not see the text "Invalid format. e.g. 01:00 PM". I checked it locally.